pkg/machinery/client/resolver/roundrobin.go

```diff
@@ -58,13 +64,35 @@ type roundRobinResolver struct {
 	port   int
 }
 
+// EnsureEndpointsHavePorts returns the list of endpoints with default port appended to those addresses that don't have a port.
+func EnsureEndpointsHavePorts(endpoints []string, defaultPort int) []string {
+	return slices.Map(endpoints, func(endpoint string) string {
+		_, _, err := net.SplitHostPort(endpoint)
+		if err != nil {
+			return net.JoinHostPort(endpoint, strconv.Itoa(defaultPort))
+		}
+
+		return endpoint
+	})
+}
+
 func (r *roundRobinResolver) start() error {
 	var addrs []resolver.Address //nolint:prealloc
 
-	for _, a := range strings.Split(r.target.Endpoint, ",") { //nolint:staticcheck
+	//nolint:staticcheck
+	endpoints := strings.Split(r.target.Endpoint, ",")
+
+	for _, addr := range endpoints {
+		serverName := addr
+
+		host, _, err := net.SplitHostPort(serverName)
+		if err == nil {
+			serverName = host
+		}
+
 		addrs = append(addrs, resolver.Address{
-			ServerName: a,
-			Addr:       fmt.Sprintf("%s:%d", net.FormatAddress(a), r.port),
+			ServerName: serverName,
+			Addr:       sideronet.FormatAddress(addr),
```

Comment:
please, please, please, don't use this method. it's not needed here, and it's going to break any correctly formatted endpoint.